src/js/components/OptionalPortsAndServiceDiscoveryComponent.jsx

```diff
@@ -246,7 +257,11 @@ var OptionalPortsAndServiceDiscoveryComponent = React.createClass({
   },
 
   render: function () {
-    if (isTooComplexStructure(this.props.fields)) {
+    var hasPlaceIdSet = PluginComponentStore.hasComponentAtPlaceId(
```

Comment:
This var name is kinda odd, what about something like `hasComponent` of `hasPluginComponent`?